connect-inject/health_check_resource_test.go

```diff
@@ -432,6 +461,115 @@ func TestReconcile_IgnorePodsWithoutInjectLabel(t *testing.T) {
 	require.Nil(actual)
 }
 
+// Test pod statuses that the reconciler should ignore.
```

Comment:
Open to opinion on this test case. I logged all the state transitions while developing this feature so thought we may as well encode that in a test case.